src/libraries/Silk.Net.Windowing/Abstract/IDesktopSurface.cs

```diff
@@ -0,0 +1,150 @@
+﻿using Silk.NET.Core;
+using Silk.NET.Maths;
+
+namespace Silk.NET.Windowing
+{
+    /// <summary>
+    /// A surface which wraps a Desktop Window.
+    /// </summary>
+    public interface IDesktopSurface : ISurface
+    {
+        /// <summary>
+        /// Whether or not the window is visible.
+        /// </summary>
+        bool IsVisible { get; set; }
+
+        /// <summary>
+        /// The position of the window. If set to -1, use the backend default.
+        /// </summary>
+        Vector2D<int> Position { get; set; }
+
+        /// <summary>
+        /// The size of the window in pixels.
+        /// </summary>
+        new Vector2D<int> Size { get; set; }
+
+        /// <summary>
+        /// The window title.
+        /// </summary>
+        string Title { get; set; }
+
+        /// <summary>
+        /// The window state.
+        /// </summary>
+        WindowState WindowState { get; set; }
+
+        /// <summary>
+        /// The window border.
+        /// </summary>
+        WindowBorder WindowBorder { get; set; }
+
+        /// <summary>
+        /// The video mode.
+        /// </summary>
+        VideoMode VideoMode { get; set; }
+    
+        /// <summary>
+        /// Gets the screen on which this window is active.
+        /// </summary>
+        IScreen? CurrentScreen { get; set; }
+
+        /// <summary>
+        /// Gets the available screens for this surface.
+        /// </summary>
+        IEnumerable<IScreen>? AvailableScreens { get; }
+
+        /// <summary>
+        /// Gets or sets whether the window waits for an event to be posted before existing <see cref="DoEvents" />.
+        /// </summary>
+        bool IsEventDriven { get; set; }
```

Comment:
Given the update loop is very similar to 2.X this will still be needed for applications (e.g. UI) that do not regularly need to render (i.e. only when an animation changes or some other event happens such as the user moving their mouse or the window etc)